Polygonal_surface_reconstruction/include/CGAL/Polygonal_surface_reconstruction/internal/point_set_with_planes.h

```diff
@@ -136,29 +136,23 @@ namespace CGAL {
                                 Base_class::resize(points.size());
                                 Base_class::add_normal_map();
 
+                                std::map<int, std::size_t> plane_index_remap;
+
                                 // Gets to know the number of plane from the plane indices
                                 int max_plane_index = 0;
-                                for (typename PointRange::const_iterator it = points.begin(); it != points.end(); ++it) {
-                                        int plane_index = get(plane_index_map, *it);
-                                        if (plane_index > max_plane_index)
-                                                max_plane_index = plane_index;
-                                }
-                                std::size_t num_plane = max_plane_index + 1; // the first one has index 0
-
-                                for (std::size_t i = 0; i < num_plane; ++i)
-                                        planar_segments_.push_back(new Planar_segment(this));
-
                                 std::size_t idx = 0;
                                 for (typename PointRange::const_iterator it = points.begin(); it != points.end(); ++it) {
                                         Base_class::m_points[idx] = get(point_map, *it);
                                         Base_class::m_normals[idx] = get(normal_map, *it);
                                         int plane_index = get(plane_index_map, *it);
                                         if (plane_index != -1) {
-                                                Planar_segment* ps = planar_segments_[plane_index];
-                                                ps->push_back(idx);
+                                          if (plane_index_remap.find(plane_index) == plane_index_remap.end()) {
```

Comment:
you should use find to get the iterator and use the iterator to update the value (instead of 2 queries

Reply:
3 queries actually

Reply:
Should be resolved with the second commit